ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,590 @@
+  │                     ->  Bitmap Heap Scan on campaigns_102046 campaigns  (cost=4.16..9.50 rows=2 width=104) │
+  │                           Recheck Cond: (company_id = 5)                                                   │
+  │                           ->  Bitmap Index Scan on campaigns_pkey_102046  (cost=0.00..4.16 rows=2 width=0) │
+  │                                 Index Cond: (company_id = 5)                                               │
+  └────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+
+Updates work too. Let's double the budget for all campaigns.
+
+::
+
+  UPDATE campaigns
+  SET monthly_budget = monthly_budget*2
+  WHERE company_id = 5;
+
+In all these queries, the filter routes SQL execution directly inside a worker. Full SQL support is available once queries are "pushed down" to a worker like this. When people scale applications with NoSQL databases they suffer the lack of transactions and joins. However per-tenant transactions work fine in Citus:
+
+::
+
+  -- transactionally remove campaign 46 and all its ads
+
+  BEGIN;
+  DELETE from campaigns where id = 46 AND company_id = 5;
+  DELETE from ads where campaign_id = 46 AND company_id = 5;
+  COMMIT;
+
+As a final demonstration of SQL support, this query includes a joins, grouping, aggregation and a window function, and it executes fine. The query ranks the ads in each campaign by the count of their impressions.
+
+::
+
+  SELECT a.campaign_id,
+         RANK() OVER (
+           PARTITION BY a.campaign_id
+           ORDER BY a.campaign_id, count(*) desc
+         ), count(*) as n_impressions, a.id
+    FROM ads as a,
+         impressions as i
+   WHERE a.company_id = 5
+     AND i.company_id = a.company_id
+     AND i.ad_id      = a.id
+  GROUP BY a.campaign_id, a.id
+  ORDER BY a.campaign_id, n_impressions desc;
+
+In short when queries are scoped to a tenant then inserts, updates, deletes, complex SQL, and transactions all work as expected.
+
+Sharing Data Between Tenants
+----------------------------
+
+Up until now all tables have been distributed by :code:`company_id`, but sometimes there is data that can be shared by all tenants, and doesn't "belong" to any tenant in particular. For instance, all companies using this example ad platform might want to get geographical information for their audience based on IP addresses. In a single machine database this could be accomplished by a lookup table for geo-ip, like the following. (A real table would probably use PostGIS but bear with the simplified example.)
+
+::
+
+  CREATE TABLE geo_ips (
+    addrs cidr NOT NULL PRIMARY KEY,
+    latlon point NOT NULL
+      CHECK (-90  <= latlon[0] AND latlon[0] <= 90 AND
+             -180 <= latlon[1] AND latlon[1] <= 180)
+  );
+  CREATE INDEX ON geo_ips USING gist (addrs inet_ops);
+
+Joining clicks with this table can produce, for example, the locations of everyone who clicked on ad 456.
+
+::
+
+  SELECT latlon
```

Comment:
I ran this query just after creating the reference table and received "ERROR: cannot plan queries that include both regular and partitioned relations". Could we make users run this query after they've run the create_reference_table() command to avoid them running into this error?